pkg/controller/provider/container/vsphere/model.go

```diff
@@ -376,14 +376,59 @@ func (v *HostAdapter) Apply(u types.ObjectUpdate) {
 					v.model.HostScsiDisks = nil
 					for _, iScsiLun := range array.ScsiLun {
 						hostScsiDisk := model.HostScsiDisk{}
-						scsiLun := iScsiLun.GetScsiLun()
-						hostScsiDisk.CanonicalName = scsiLun.CanonicalName
-						hostScsiDisk.Vendor = strings.TrimSpace(scsiLun.Vendor)
-						v.model.HostScsiDisks = append(v.model.HostScsiDisks, hostScsiDisk)
+						if disk, ok := iScsiLun.(*types.HostScsiDisk); ok {
+							hostScsiDisk.CanonicalName = disk.CanonicalName
+							hostScsiDisk.Vendor = strings.TrimSpace(disk.Vendor)
+							hostScsiDisk.Model = strings.TrimSpace(disk.Vendor)
```

Comment:
Unsure we really need to collect a model here.

Reply:
This was leftover, I removed it, thank you for the catch